src/aliceVision/sfm/LocalBundleAdjustmentGraph.cpp

```diff
@@ -657,24 +657,26 @@ std::size_t LocalBundleAdjustmentGraph::addIntrinsicEdgesToTheGraph(const sfmDat
 
     for(const auto& x : _nodePerViewId) // for each reconstructed view in the graph
     {
-      if(newViewId == x.first)  // do not compare a view with itself
-        continue;
-      
-      const IndexT reconstructedViewIntrinsicId = sfmData.getViews().at(x.first)->getIntrinsicId();
-      
+      const auto& otherViewId = x.first;
+
       // if the new view share the same intrinsic than previously reconstructed views
-      if(reconstructedViewIntrinsicId == newViewIntrinsicId)
+      if(newViewId != otherViewId  // do not compare a view with itself
```

Comment:
`if(newViewId != otherViewId  // do not compare a view with itself (as "_nodePerViewId" already contains the "newReconstructedViews")`